src/metrics/builder.rs

```diff
@@ -0,0 +1,117 @@
+use goodmetrics::{
+    default_gauge_factory,
+    downstream::{get_client, OpenTelemetryDownstream, OpentelemetryBatcher},
+    pipeline::DimensionPosition,
+};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::mpsc;
+use tokio_rustls::rustls::RootCertStore;
+use tonic::metadata::MetadataValue;
+
+use super::{proxy::DefaultProxyMetrics, util::proxy_sum_gauge, RpcMetrics};
+
+pub struct ProxyMetricsBuilder {
+    batch_interval: Duration,
+    batch_capacity: usize,
+}
+
+impl ProxyMetricsBuilder {
+    pub fn new() -> Self {
+        Self {
+            batch_interval: Duration::from_secs(1),
+            batch_capacity: 128,
+        }
+    }
+
+    pub async fn build(self) -> Arc<DefaultProxyMetrics> {
+        let (batch_sender, batch_receiver) = mpsc::channel(self.batch_capacity);
+        let gauge_factory = default_gauge_factory();
+
+        let endpoint = get_environment_variable_or_none("OTLP_ENDPOINT");
+        let api_token = get_environment_variable_or_none("OTLP_API_TOKEN");
+
+        if endpoint.is_some() && api_token.is_some() {
```

Comment:
fyi in case you're not aware, you can do this without an unwrap:
```rust
match (&endpoint, &api_token) {
  (Some(endpoint), Some(api_token)) => {
    // do setup
  }
  (None, _) => {
    // log missing endpoint
  }
  (_, None) => {
    // log missing token
  }
}
```

Reply:
I was not aware. Thank you! Applied in [1a8a55a](https://github.com/momentohq/momento-proxy/pull/4/commits/1a8a55a016032392b144f0068f021ea21f754f12)